workflow-templates/shellcheck.yml

```diff
@@ -0,0 +1,56 @@
+# Run ShellCheck on PRs and $default-branch
+
+# For more context, see:
+# https://github.com/openedx/.github/blob/master/docs/decisions/0001-shellcheck.rst
+
+name: ShellCheck
+
+on:
+  pull_request:
+  push:
+    branches:
+    - $default-branch
+
+permissions:
+  contents: read
+
+jobs:
+  shellcheck:
+    runs-on: ubuntu-latest
```

Comment:
This check only runs on Ubuntu.

I was tempted to generalize it to MacOS. It would be easy enough to use a build matrix here so that this runs on both ubuntu-latest and macos-latest. What would be a little more difficult is ensuring that we download the correct binary in action.yml.

I decided against that because I've already been at this for a few days, and I feel like I could spend another week tweaking and perfecting everything, but it's not clear to me that it'd be time well spent. If you disagree, let me know.